source/views/streaming/streams/list.tsx

```diff
@@ -84,38 +64,33 @@ export const StreamListView = (): JSX.Element => {
 					$groupBy: group,
 				}
 			})
-
-		let grouped = groupBy(data, (j) => j.$groupBy)
-		let mapped = toPairs(grouped).map(([title, data]) => ({title, data}))
-
-		setStreams(mapped)
-	}
-
-	let keyExtractor = (item: StreamType) => item.eid
-
-	let renderItem = ({item}: {item: StreamType}) => <StreamRow stream={item} />
-
-	if (loading) {
-		return <LoadingView />
-	}
+	}, [data])
 
 	if (error) {
-		return <NoticeView text={`Error: ${error.message}`} />
+		return (
+			<NoticeView
+				buttonText="Try Again"
+				onPress={reload}
+				text={`A problem occured while loading the streams. ${error.message}`}
+			/>
+		)
 	}
 
 	return (
 		<SectionList
 			ItemSeparatorComponent={ListSeparator}
-			ListEmptyComponent={<NoticeView text="No Streams" />}
+			ListEmptyComponent={
+				isLoading ? <LoadingView /> : <NoticeView text="No streams." />
+			}
 			contentContainerStyle={styles.contentContainer}
-			keyExtractor={keyExtractor}
-			onRefresh={refresh}
-			refreshing={refreshing}
-			renderItem={renderItem}
+			keyExtractor={(item: StreamType) => item.eid}
+			onRefresh={reload}
+			refreshing={isPending && !isInitial}
+			renderItem={({item}: {item: StreamType}) => <StreamRow stream={item} />}
```

Comment:
Do you need the type annotation here, or is it smart enough to understand the following?

```suggestion
			renderItem={({item}) => <StreamRow stream={item} />}
```

Reply:
It is smart enough to understand that. Good suggestion.